plugins/pulp_rpm/plugins/importers/yum/modularity.py

```diff
@@ -0,0 +1,284 @@
+import os
+
+from hashlib import sha256
+from uuid import uuid4
+
+from mongoengine import Q, NotUniqueError
+
+from pulp.server.controllers import repository as repository_controller
+from pulp.server.managers.repo._common import get_working_directory
+
+from pulp_rpm.plugins.db.models import Modulemd, ModulemdDefaults, RPM
+from pulp_rpm.plugins.importers.yum.repomd import modules
+from pulp_rpm.plugins.importers.yum.parse import rpm
+
+
+def get_inventory(repository, Model):
+    """
+    Get content that is already contained in the repository.
+
+    :param repository: A repository.
+    :type repository: pulp.server.db.model.Repository
+    :param Model: The content model class.
+    :type Model: pulp.server.db.model.FileContent
+    :return: A dict keyed by the unit key tuple and a value of checksum.
+    """
+    inventory = {}
+    fields = ('id', 'checksum')
+    fields += Model.unit_key_fields
+    q_set = repository_controller.find_repo_content_units(
+        repository,
+        repo_content_unit_q=Q(unit_type_id=Model.TYPE_ID),
+        unit_fields=fields,
+        yield_content_unit=True)
+    for m in q_set:
+        key = m.NAMED_TUPLE(**m.unit_key)
+        inventory[key] = m.checksum
+    return inventory
+
+
+def add_modulemd(repository, modulemd, model):
+    """
+    Add the specified modulemd.
+
+    Created as needed then added to the repository.
+
+    :param repository: A repository.
+    :type repository: pulp.server.db.model.Repository
+    :param modulemd: A modulemd (libmodulemd) document object.
+    :type modulemd: gi.repository.Modulemd.Module
+    :param model: The Modulemd model object to add.
+    :type model: pulp_rpm.plugins.db.models.Modulemd
+    """
+    path = os.path.join(
+        get_working_directory(),
+        str(uuid4()))
+    document = modulemd.dumps()
```

Comment:
i was wondering if we could use modulemd.dump(model._storage_path) so it would save the yaml file directly on the disk in the location we specified
https://sgallagh.fedorapeople.org/docs/libmodulemd/1.5/modulemd-1.0-modulemd-module.html#modulemd-module-dump